shared/src/main/kotlin/serialization/Json.kt

```diff
@@ -0,0 +1,107 @@
+ * limitations under the License.
+ */
+
+package ai.tock.shared.serialization
+
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.InternalSerializationApi
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerializationStrategy
+import kotlinx.serialization.builtins.ListSerializer
+import kotlinx.serialization.builtins.SetSerializer
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeFromStream
+import kotlinx.serialization.json.encodeToStream
+import kotlinx.serialization.modules.SerializersModule
+import kotlinx.serialization.modules.contextual
+import kotlinx.serialization.serializer
+import mu.KotlinLogging
+import java.io.BufferedReader
+import java.io.InputStream
+import java.io.OutputStream
+
+@OptIn(ExperimentalSerializationApi::class)
+@PublishedApi
+internal val json =
+    Json {
+        prettyPrint = false
+        isLenient = true
+        ignoreUnknownKeys = true
+        coerceInputValues = true
+        explicitNulls = false
+        serializersModule =
+            SerializersModule {
+                contextual(AnySerializer())
+            }
+    }
+
+class AnySerializer : KSerializer<Any> {
+    override val descriptor: SerialDescriptor = buildClassSerialDescriptor("Any")
+
+    @Suppress("UNCHECKED_CAST")
+    @OptIn(InternalSerializationApi::class)
+    override fun serialize(
+        encoder: Encoder,
+        value: Any,
+    ) {
+        when (value) {
+            is List<*> -> {
+                val listSerializer = ListSerializer(AnySerializer())
+                encoder.encodeSerializableValue(listSerializer as SerializationStrategy<Any>, value)
+            }
+
+            is Set<*> -> {
+                val setSerializer = SetSerializer(AnySerializer())
+                encoder.encodeSerializableValue(setSerializer as SerializationStrategy<Any>, value)
+            }
+
+            else -> {
+                encoder.encodeSerializableValue(value::class.serializer() as SerializationStrategy<Any>, value)
+            }
+        }
+    }
+
+    override fun deserialize(decoder: Decoder): Any = error("no serialization class available")
+}
+
+@OptIn(ExperimentalSerializationApi::class)
+inline fun <reified T : Any> readJson(input: InputStream): T = json.decodeFromStream<T>(input)
+
+inline fun <reified T : Any> readJson(content: String): T = json.decodeFromString(content)
+
+inline fun <reified T> loadResource(path: String): T? =
+    object {}::class.java.getResourceAsStream(path)?.let { stream ->
+        readJson(
+            stream
+                .bufferedReader()
+                .use(BufferedReader::readText),
```

Comment:
Why not call `readJson(InputStream)` here instead of loading the whole string first?